### PR TITLE
Refactor hist normalization, remove assert from library code

### PR DESCRIPTION
**Pull request by maxnoe**
I saw we had a defensive assert in the actual code, which is actually covered by the unit tests.

I also found a more understandable way of doing the normalization than this swapaxes stuff.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,23 +50,24 @@ jobs:
...
+          python-version: 3.8
+
+      - name: Install doc dependencies
         run: |
-          ./download_test_data.sh
+          sudo apt update --yes && sudo apt install --yes git build-essential pandoc curl graphviz
+          pip install -U pip setuptools wheel
+          pip install -e .[docs]
+          git describe --tags
+          python -c 'import ctapipe; print(ctapipe.__version__)'
```
Comment (HealthyPear) [inline .github/workflows/ci.yml:67]: `ctapipe` should be changed in `pyirf` (this is also the reason why the `docs` part of the CI is failing)

**Review by HealthyPear — CHANGES_REQUESTED** [review]
Code is fine, there is only a fix to be done in the CI

File: .github/workflows/ci.yml (hunk 1 above)
Comment (maxnoe) [reply .github/workflows/ci.yml:67]: thx

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -50,23 +50,24 @@ jobs:
...
-        with:
-          docs-folder: "docs/"
-          pre-build-command: |
-            apt update --yes && apt install --yes git build-essential pandoc curl
-            pip install -U pip setuptools wheel sphinx
-            pip install -e '.[docs]'
-            python examples/calculate_eventdisplay_irfs.py
-          build-command: make html SPHINXOPTS="-W --keep-going -n --color -w /tmp/sphinx-log"
+      - name: Build docs
+        run: make doc SPHINXOPTS="-W --keep-going -n --color -j auto"
```
Comment (HealthyPear) [inline .github/workflows/ci.yml:70]: sorry I overlooked this: we don't have a MakeFile with the definition for `doc`, so just `make html` will work

**Review by HealthyPear — CHANGES_REQUESTED** [review]

**Comment by maxnoe** [comment]
@HealthyPear Sorry I let this slip for so long, it should be fine now

File: setup.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import os
 
 
-gammapy = "gammapy~=0.18"
+gammapy = "gammapy~=0.18.0"
```
Comment (HealthyPear) [inline setup.py:5]: Not a problem, but what's the need for this variable substitution?

Comment (maxnoe) [reply setup.py:5]: We have `gammapy` in three optional dependencies. This is to make sure, we use the same version everywhere. I just forgot it at one point (used the variable in 2 and the string in another one)

**Review by HealthyPear — APPROVED** [review]

**Pull request merged**